lib/parser.js

```diff
@@ -252,14 +257,26 @@ module.exports = {
       var h = parsed.params.headers[i].toLowerCase();
       parsed.params.headers[i] = h;
 
-      if (h !== 'request-line') {
+      if (h == 'request-line') {
+        if (!options.strict) {
+          /**
+           * We allow headers from the older spec drafts if strict parsing isn't
+           * specified in options.
+           */
+          parsed.signingString +=
+            request.method + ' ' + request.url + ' HTTP/' + request.httpVersion;
+        } else {
+          /* Strict parsing doesn't allow older draft headers. */
+          throw new StrictParsingError('request-line is not a valid header with strict parsing enabled.');
+        }
+      } else if(h == '(request-target)') {
+        parsed.signingString +=
+          '(request-target): ' + request.method.toLowerCase() + ' ' + request.url;
+      } else {
         var value = request.headers[h];
         if (!value)
```

Comment:
This check should probably be against `undefined` instead of a general truthiness check. This code will fail, for instance, if the value is the empty string. While that may be ill-advised as an http header value, it is valid.